pkg/detectors/openai/openai.go

```diff
@@ -20,7 +20,8 @@ type Scanner struct{}
 var _ detectors.Detector = (*Scanner)(nil)
 
 var (
-	keyPat = regexp.MustCompile(`\b((?:sk)-[a-zA-Z0-9]{48})\b`)
+	// The magic string T3BlbkFJ is the base64-encoded string: OpenAI
+	keyPat = regexp.MustCompile(`\b((?:sk)-[a-zA-Z0-9]{20}T3BlbkFJ[a-zA-Z-09]{20})\b`)
```

Comment:
suggestion: Looks like a typo in the last character class, it should be `[a-zA-Z-0-9]` looks like we are missing a `-`

question: Is there a benefit in keeping the initial non-capture group? Could we remove it and get the same effect.
eg: `\b(sk-[a-zA-Z0-9]{20}T3BlbkFJ[a-zA-Z-0-9]{20})\b`